juniper_subscriptions/src/lib.rs

```diff
@@ -0,0 +1,446 @@
+//! This crate supplies [`SubscriptionCoordinator`] and
```

Comment:
This is an interesting question. 

I was thinking about providing a coordinator and a subscription type, which will cover most of the use cases, but allow users to implement their own coordinators and connections if needed. 

`Value<Stream<Item = Value>>` is returned from Juniper's resolvers, but it would be better to iterate over `GraphQLResponse`s in a server library (+ have some functionality on top of this stream). Each Object/List is assembled from items, which are returned from streams. One might want to wait until all values from `Object`/`List` are yielded and then return the complete thing, while other might want to return values as soon as they arrive. Some people might not mind returning incomplete objects, as long as they have some required fields filled up. This is the main reason for moving `SubscriptionConnection` into a separate trait.

I didn't implement a lot in `SubscriptionCoordinator` yet, but I think it should be in a separate trait as well. Some users might want to add a custom health checker to a connection, others might want to decline another connection if CPU usage is greater than 80%.

These cases will not be covered by `juniper_subscriptions`, but I was planning to cover [those cases](https://github.com/graphql-rust/juniper/pull/433#issuecomment-553570544) for now